design/snapshots.md

```diff
@@ -72,20 +72,20 @@ that, if mounted, will have the fully prepared snapshot at the requested path.
 We call this the _prepare_ operation.
 
 Once a path is _prepared_ and mounted, the caller may write new data to the
-snapshot. Depending on application, a user may want to capture these changes or
+snapshot. Depending on the application, a user may want to capture these changes or
 not.
 
-If the user wants to keep the changes, the _commit_ operation is employed.  The
+If the user wants to keep the changes, the _commit_ operation is employed. The
 _commit_ operation takes the `target` directory, which represents an open
-transaction, and a `diff` directory. A successful result will end up with the
-difference between the parent and snapshot in the `diff` directory, which
+transaction, and a `diff` directory. A successful result will provide the
+difference between the parent and the snapshot in the `diff` directory, which
 should be treated as opaque by the caller. This new `diff` directory can then
 be used as the `parent` in calls to future _prepare_ operations.
 
 If the user wants to discard the changes, the _rollback_ operation will release
```

Comment:
Indeed, this is the case. We can still merge this cleanup.

I was planning on revisiting this once we did an end to end evaluation (ie working push/pull) but feel free to start migrating the nomenclature.

There are still a few tricky bits that need to be worked out, such as parentage on commit. Those changes will be less intrusive, so this should be a lot less work.